src/python/pants/backend/cc/lint/clangformat/subsystem.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from typing import Iterable
+
+from pants.backend.python.goals import lockfile
+from pants.backend.python.goals.lockfile import GeneratePythonLockfile
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.goals.generate_lockfiles import GenerateToolLockfileSentinel
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import ArgsListOption, SkipOption
+from pants.util.docutil import git_url
+from pants.util.strutil import softwrap
+
+
+class ClangFormat(PythonToolBase):
+    options_scope = "clang-format"
```

Comment:
Bikeshed on naming: "clangformat" or "clang-format"

"ClangFormat" is the tool's name. I also named variables/packages as "clangformat" (excluding the skip field, which is `skip_clang_format` to line up with the options scope). 

"clang-format" is the executable's name and is the command-line usage. Also, configuration is a ".clang-format" file.

So the question comes down to `./pants help clang-format` vs `./pants help clangformat` essentially.